src/test/java/org/folio/rest/impl/AcquisitionsUnitAssignmentsTest.java

```diff
@@ -0,0 +1,156 @@
+
+  @Test
+  public void testGetAcqUnitAssignmentsWithQuery() {
+    logger.info("=== Test GET Acquisitions Unit Assignments - search by query ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "?query=recordId==57257531-c6bc-4b6e-b91a-06bc0034b768";
+
+    final AcquisitionsUnitAssignmentCollection units = verifySuccessGet(url, AcquisitionsUnitAssignmentCollection.class);
+    assertThat(units.getAcquisitionsUnitAssignments(), hasSize(1));
+  }
+
+  @Test
+  public void testGetAcqUnitAssignmentsWithUnprocessableQuery() {
+    logger.info("=== Test GET Acquisitions Unit Assignments - unprocessable query ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "?query=" + BAD_QUERY;
+
+    verifyGet(url, APPLICATION_JSON, 400);
+  }
+
+  @Test
+  public void testGetAcqUnitAssignmentsSuccess() {
+    logger.info("=== Test GET Acquisitions Unit Assignment - success case ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "/8ad580ee-d3f1-4271-bff4-c7c80ccb22e5";
+
+    final AcquisitionsUnitAssignment unit = verifySuccessGet(url, AcquisitionsUnitAssignment.class);
+    assertThat(unit, notNullValue());
+  }
+
+  @Test
+  public void testGetAcqUnitAssignmentNotFound() {
+    logger.info("=== Test GET Acquisitions Unit Assignment - not found ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "/" + ID_DOES_NOT_EXIST;
+
+    verifyGet(url, APPLICATION_JSON, 404);
+  }
+
+  @Test
+  public void testPutAcqUnitAssignmentSuccess() {
+    logger.info("=== Test PUT Acquisitions Unit Assignment - success case ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "/8ad510ee-d3f1-4271-bff4-c7c80ccb22e5";
+
+    verifyPut(url, JsonObject.mapFrom(new AcquisitionsUnitAssignment()
+      .withRecordId("47Ac60b4-159D-4e1c-9aCB-8293Df67D16d")
+      .withAcquisitionsUnitId("c2d6608f-6d1f-45f7-8817-5d32c2416116")), "", 204);
+  }
+
+  @Test
+  public void testValidationOnPutUnitAssignmentWithoutBody() {
+    logger.info("=== Test validation on PUT Acquisitions Unit Assignment with no body ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "/" + UUID.randomUUID().toString();
+
+    verifyPut(url, "", TEXT_PLAIN, 400);
+  }
+
+  @Test
+  public void testPutUnitAssignmentNotFound() {
+    logger.info("=== Test PUT Acquisitions Unit Assignment - not found ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "/" + ID_DOES_NOT_EXIST;
+
+    verifyPut(url, JsonObject.mapFrom(new AcquisitionsUnitAssignment()
+      .withRecordId("57257531-c6bc-4b6e-b91a-06bc0034b768")
+      .withAcquisitionsUnitId("8ad510ee-d3f1-4271-bff4-c7c80ccb22e5")), APPLICATION_JSON, 404);
+  }
+
+  @Test
+  public void testPutUnitAssignmentIdMismatch() {
+    logger.info("=== Test PUT Acquisitions Unit Assignment - different ids in path and body ===");
+    String url = ACQ_UNIT_ASSIGNMENTS_ENDPOINT + "/" + UUID.randomUUID().toString();
+
+    AcquisitionsUnitAssignment unitAssignment = new AcquisitionsUnitAssignment().withRecordId("57257531-c6bc-4b6e-b91a-06bc0034b768")
```

Comment:
This UUID `57257531-c6bc-4b6e-b91a-06bc0034b768` is used in multiple places, can be stored in `final` variable.

Reply:
Done